src/libraries/System.Private.CoreLib/src/System/Resources/ResourceReader.Core.cs

```diff
@@ -73,6 +76,12 @@ private object DeserializeObject(int typeIndex)
         }
 
         // Issue https://github.com/dotnet/runtime/issues/39290 tracks finding an alternative to BinaryFormatter
+        [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2060:MakeGenericMethod",
```

Comment:
This method should be marked as `[RequiresUnreferencedCode]`. And then `DeserializeObject` should have the suppression on it, with the justification that we have a feature switch where the unsafe method won't be called.